### ISSUE TITLE
Spouses not added to descendants

### ISSUE THREAD

**Issue by Neriderc** [Neriderc/GVExport#85]
It seems that when you choose to include "Spouses", spouses are only added to ancestors (the starting individual's generation and older). I've tested older versions and this bug exists even in the earliest version on this repository, so it's not a new bug.

Solution is to add spouses to descendants as well.

**Comment by schuco** [comment]
At least in my older version (for 2.0) there  is no such bug. Rather spouses are not added only for the last generation of ancestors. But they are added in all descendens-generations before. (See screenshot below).
Therefore if you chose to display 1 descendents generation you will not see any spouses .

This is in my opinion is no bug (if it works the same way in the actual version), as it makes sense, not to include spouses for the last descendents-generation as long as you don't see the offspring of this families.

![Firefox_Screenshot_2022-07-12T12-15-50 567Z](https://user-images.githubusercontent.com/5896925/178488769-48306ad1-4e42-4ec3-8759-8f991f4e82d7.png)
.

**Comment by Neriderc** [comment]
Thanks for this, you're right! I didn't play with it enough, it does just seem to be the last generation. It does make sense that if we aren't adding the families then the spouses shouldn't be there, except in the "combined" diagram type, and it seems to correctly add them in that case.

On a slightly related note, in the older version, if you choose a starting individual and choose not to descendants, but then you select "cousins", do you find that the nieces and nephews are included? Is this a feature or a bug? My expectation was that if "Include descendants" is turned off, there should not be any generation listed that is younger than the starting individual.

**Comment by schuco** [comment]
Your question:
"On a slightly related note, in the older version, if you choose a starting individual and choose not to include descendants, but then you select "cousins", do you find that the nieces and nephews are included? Is this a feature or a bug? My expectation was that if "Include descendants" is turned off, there should not be any generation listed that is younger than the starting individual."

In my version (for wt2.0) there is no such bug or feature. As you expect if "Include descendants" is turned off, there is not any generation listed that is younger than the starting individual.

To prove that and to demonstrate how GVExport collects the individuals to be shown with different option around the cousin-parameter, I add some examples with screenshots. The term "cousin" means that for all persons to be displayed in the diagram the descendants of sibs in the parent's generation are shown (as long as they fit to the selected number of generations and descendants are to be shown.
(There was a debate on cousins in a former, closed issue. In my opinion the label cousin would be o.k. as it describes how GVExport collects the individuals to be shown. I admit that the term cousin is not correct when you have a relation by adoption.)

From my definition of what "cousin" means, it follows that you will not see any cousin-relation of the starting individual (nor from anyone else) if you select 1 ancestor generation. As the grandparent-generation is not to be displayed there are no sibs of the parents displayed, consequently their children - which would be cousins of the starting individual - are missing. Parameter "with or without sibs" or "with or without descendants" have no effect.

Example 1 with options: 1 ancestor generation, display sibs, cousins, spouses and descendants (1 generation) 
No  cousins to be seen, though cousins is selected. Reason: Grandparents not to be displayed (see above).
![#6_1AxSibxCouxSpoxDesc1D_Cousin added_Descendands added](https://user-images.githubusercontent.com/5896925/178824105-bbc7ab47-36c3-4e3f-b94c-c257e8599f21.png)

Example 2 with options: 2 ancestor generation, display sibs, cousins, spouses and descendants (1 generation)
Cousins to be seen as grandparent generation is to be displayed
![#7_2AxSibxCouxSpoxDesc1D_Cousin added_Descendands added-2 Gen Ancestors](https://user-images.githubusercontent.com/5896925/178824406-8259e699-24ed-4a5e-91a2-62032574253b.png)

Example 3 with options: 2 ancestor generation, display cousins, spouses and descendants (1 generation), NO SIBS
No  cousins to be seen, though 2 ancestor generations and cousins is selected. Reason NO SIBS of the parents to be displayed (These would be parents of cousins)
![#8_2A-SibxCouxSpoxDesc1D_Cousin added_Descendands added-2 Gen Ancestors-no sibs](https://user-images.githubusercontent.com/5896925/178824793-dcc66ae5-9bb7-496c-b223-3268be408fe5.png)

Example 4 with NO DESCENDANTS to be displayed, no sibs, cousins to be displayed
This is to show that if you choose not to include descendants, but then you select "cousins", you don't find that the nieces and nephews are included
![#9_2A-SibxCouxSpo-Desc0D_Cousin added_-2 Gen Ancestors-no sibs-no ancest](https://user-images.githubusercontent.com/5896925/178826635-47774a17-91f4-4bfd-8c69-fca5eaf7ea37.png)

**Comment by Neriderc** [comment]
Thanks but I don't think any of your examples show the situation I'm talking about, which is with "Cousins" _and_ "Siblings" checked, but "Include Descendants" unchecked.

Here's an example using one of Shakespeare's kids, Susanna. Here she is with 1 generation of ancestors (her parents), one generation of descendants (her daughter + nephews), and in this case we have "Siblings", "Cousins", and "Spouses" all selected:
![image](https://user-images.githubusercontent.com/89974636/178843449-fe706fe6-660e-4797-990a-e7c43ad75bef.png)

Now if I _remove_ the descendants, here's what I see:
![image](https://user-images.githubusercontent.com/89974636/178843632-9a4665e8-19c3-405d-95ef-95756edd84db.png)

I still have the nephews included, but I unchecked "Include descendants", so I expect this would also not include descendants of siblings (it also includes descendants of cousins).

On a side note, I notice you don't have the issue of text running outside the box. This has always been an issue for me, and I presumed this was an issue with the tool used to generate the diagram in the browser. Seeing it works fine for you makes me reconsider this assumption.

**Comment by schuco** [comment]
I modified my example 4 to check sibs, as in your Shakespear drama.
There would be lots of cousins for Theodor Heinrich Kamphaus available, but non of them is displayed. So in my example there is no bug to be seen.
![#10_2AxSibxCouxSpo-Desc0D_sibs and cousin added_-2 Gen Ancestors-no ancest](https://user-images.githubusercontent.com/5896925/179013387-bb6f0c1b-17a1-43c8-b43f-228914f13f9e.png)

It is true that text running out of the box is not seen in my example. However I have noticed it in my version too. The reason why its less often seen in my case may be that the names are longer due to the PID which I can't avoid with Vesta classic look and feel.
